pkg/resources/table.go

```diff
@@ -0,0 +1,287 @@
+	if err != nil {
+		return "", err
+	}
+	strTableID := strings.TrimSpace(buf.String())
+	return strTableID, nil
+}
+
+// tableIDFromString() takes in a pipe-delimited string: DatabaseName|SchemaName|TableName
+// and returns a tableID object
+func tableIDFromString(stringID string) (*tableID, error) {
+	reader := csv.NewReader(strings.NewReader(stringID))
+	reader.Comma = tableIDDelimiter
+	lines, err := reader.ReadAll()
+	if err != nil {
+		return nil, fmt.Errorf("Not CSV compatible")
+	}
+
+	if len(lines) != 1 {
+		return nil, fmt.Errorf("1 line at a time")
+	}
+	if len(lines[0]) != 3 {
+		return nil, fmt.Errorf("3 fields allowed")
+	}
+
+	tableResult := &tableID{
+		DatabaseName: lines[0][0],
+		SchemaName:   lines[0][1],
+		TableName:    lines[0][2],
+	}
+	return tableResult, nil
+}
+
+// CreateTable implements schema.CreateFunc
+func CreateTable(data *schema.ResourceData, meta interface{}) error {
+	db := meta.(*sql.DB)
+	database := data.Get("database").(string)
+	schema := data.Get("schema").(string)
+	name := data.Get("name").(string)
+
+	// This type conversion is due to the test framework in the terraform-plugin-sdk having limited support
+	// for data types in the HCL2ValueFromConfigValue method.
+	columns := []map[string]string{}
+	for _, column := range data.Get("column").([]interface{}) {
+		columnDef := map[string]string{}
+		for key, val := range column.(map[string]interface{}) {
+			columnDef[key] = val.(string)
+		}
+		columns = append(columns, columnDef)
+	}
+	builder := snowflake.TableWithColumnDefinitions(name, database, schema, columns)
+
+	// Set optionals
+	if v, ok := data.GetOk("comment"); ok {
+		builder.WithComment(v.(string))
+	}
+
+	stmt := builder.Create()
+	err := snowflake.Exec(db, stmt)
+	if err != nil {
+		return errors.Wrapf(err, "error creating table %v", name)
+	}
+
+	tableID := &tableID{
+		DatabaseName: database,
+		SchemaName:   schema,
+		TableName:    name,
+	}
+	dataIDInput, err := tableID.String()
+	if err != nil {
+		return err
+	}
+	data.SetId(dataIDInput)
+
+	return ReadTable(data, meta)
+}
+
+// ReadTable implements schema.ReadFunc
+func ReadTable(data *schema.ResourceData, meta interface{}) error {
+	db := meta.(*sql.DB)
+	tableID, err := tableIDFromString(data.Id())
+	if err != nil {
+		return err
+	}
+
+	dbName := tableID.DatabaseName
+	schema := tableID.SchemaName
+	name := tableID.TableName
+
+	stmt := snowflake.Table(name, dbName, schema).Show()
+	row := snowflake.QueryRow(db, stmt)
+	table, err := snowflake.ScanTable(row)
+	if err != nil {
+		return err
+	}
+
+	err = data.Set("name", table.TableName.String)
+	if err != nil {
+		return err
+	}
+
+	err = data.Set("owner", table.Owner.String)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// UpdateTable implements schema.UpdateFunc
+func UpdateTable(data *schema.ResourceData, meta interface{}) error {
+	// https://www.terraform.io/docs/extend/writing-custom-providers.html#error-handling-amp-partial-state
+	data.Partial(true)
```

Comment:
If this goes in after https://github.com/chanzuckerberg/terraform-provider-snowflake/pull/262, we need to remove partial usage.